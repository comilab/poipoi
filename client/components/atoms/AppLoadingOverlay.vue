<template lang="pug">
div
  .absolute.inset-0.h-full(v-if="loading", data-testid="overlay")
    .position-center
      font-awesome-icon.fa-6x.text-black.opacity-50(
        :icon="faSpinner",
        pulse
      )
  slot(v-else)
</template>

<script lang="ts">
import { defineComponent } from 'nuxt-composition-api'
import { faSpinner } from '@fortawesome/free-solid-svg-icons'

export default defineComponent({
  props: {
    loading: {
      type: Boolean,
      required: true
    }
  },
  setup () {
    return {
      faSpinner
    }
  }
})
</script>
