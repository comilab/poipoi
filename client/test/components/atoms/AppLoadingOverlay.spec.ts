import { render, RenderOptions } from '@testing-library/vue'

import AppLoadingOverlay from '~/components/atoms/AppLoadingOverlay.vue'

describe('components/atoms/AppLoadingOverlay', () => {
  let options: RenderOptions<AppLoadingOverlay>

  beforeEach(() => {
    options = {
      slots: {
        default: 'slot'
      },
      stubs: ['FontAwesomeIcon'],
      props: {
        loading: true
      }
    }
  })

  test('loading = true ならoverlayが表示される', () => {
    const { container, getByTestId, queryByText } = render(AppLoadingOverlay, options)

    expect(container).toMatchSnapshot()
    expect(getByTestId('overlay')).toBeVisible()
    expect(queryByText('slot')).not.toBeInTheDocument()
  })

  test('loading = false ならslotが表示される', async () => {
    const { updateProps, container, queryByTestId, getByText } = render(AppLoadingOverlay, options)

    await updateProps({ loading: false })

    expect(container).toMatchSnapshot()
    expect(queryByTestId('overlay')).not.toBeInTheDocument()
    expect(getByText('slot')).toBeVisible()
  })
})
