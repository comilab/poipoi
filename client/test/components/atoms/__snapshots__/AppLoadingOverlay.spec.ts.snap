// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/AppLoadingOverlay loading = false ならslotが表示される 1`] = `
<div>
  <div>
    slot
  </div>
</div>
`;

exports[`components/atoms/AppLoadingOverlay loading = true ならoverlayが表示される 1`] = `
<div>
  <div>
    <div
      class="absolute inset-0 h-full"
      data-testid="overlay"
    >
      <div
        class="position-center"
      >
        <fontawesomeicon-stub
          class="fa-6x text-black opacity-50"
          icon="[object Object]"
          pulse=""
        />
      </div>
    </div>
  </div>
</div>
`;
